build_tools/kickstarts/cert-gen.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+# Generate certs pair for configuring TLS enabled docker daemon
+. /etc/sysconfig/docker
+
+if [ -e $DOCKER_CERT_PATH/ca.pem ]; then
+  # Certificates already generated
+  exit
+fi
+
+function randomString {
+        # if a param was passed, it's the length of the string we want
+        if [[ -n $1 ]] && [[ "$1" -lt 20 ]]; then
+                local myStrLength=$1;
+        else
+                # otherwise set to default
+                local myStrLength=8;
+        fi
+
+        dd if=/dev/urandom bs=1 2>/dev/null | tr -dc '[:alnum:]' | dd bs=1 count=$myStrLength 2>/dev/null
+}
+
+# Get a temporary workspace
+dir=`mktemp -d`
+cd $dir
+
+# Get a random password for the CA and save it
+passfile=tmp.pass
+password=$(randomString 10)
+echo $password > $passfile
+
+# Generate the CA
+openssl genrsa -aes256 -passout file:$passfile -out ca-key.pem 2048
+openssl req -new -x509 -passin file:$passfile -days 365 -key ca-key.pem -sha256 -out ca.pem -subj "/C=/ST=/L=/O=/OU=/CN=example.com"
+
+# Generate Server Key and Sign it
+openssl genrsa -out server-key.pem 2048
+openssl req -subj "/CN=example.com" -new -key server-key.pem -out server.csr
+# Allow from routable local IP
+extip=`ip addr show eth1 2>/dev/null | awk 'NR==3 {print $2}' | cut -f1 -d\/`
+if [[ -z "$extip" ]]; then
+      extip=`ip route get 8.8.8.8 | awk 'NR==1 {print $NF}'`
+fi
+extipfile=extfile.cnf
+echo subjectAltName = IP:$extip > $extipfile
```

Comment:
But ... this detection is failing because the network interfaces come up late :(

==> default: Importing base box 'adb-scratch'...
==> default: Matching MAC address for NAT networking...
==> default: Setting the name of the VM: vagrant-adbinfo_default_1448281079450_48669
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
    default: Adapter 2: hostonly
==> default: Forwarding ports...
    default: 22 => 2222 (adapter 1)
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Connection timeout. Retrying...
    default: 
    default: Vagrant insecure key detected. Vagrant will automatically replace
    default: this with a newly generated keypair for better security.
    default: 
    default: Inserting generated public key within guest...
    default: Removing insecure key from the guest if its present...
    default: Key inserted! Disconnecting and reconnecting using new SSH key...
==> default: Machine booted and ready!
==> default: Checking for guest additions in VM...
    default: No guest additions were detected on the base box for this VM! Guest
    default: additions are required for forwarded ports, shared folders, host only
    default: networking, and more. If SSH fails on this machine, please install
    default: the guest additions and repackage the box to continue.
    default: 
    default: This is not an error message; everything may continue to work properly,
    default: in which case you may ignore this message.
==> default: Configuring and enabling network interfaces...
==> default: Rsyncing folder: /home/bexelbie/Repositories/vagrant-adbinfo/ => /vagrant